### ISSUE TITLE
Bug: prefer-destructring should not trigger on array indices > 0

### ISSUE THREAD

**Issue by cgarrovillo** [eslint/eslint#16043]
### Environment

Node version: 18.4.0
npm version: 8.12.2
Local ESLint version: 8.18.0
Global ESLint version: N/A
Operating System: MacOS 12.2.1

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
{
    "parserOptions": {
        "ecmaVersion": "latest",
        "sourceType": "script",
        "ecmaFeatures": {}
    },
    "rules": {
        "prefer-destructuring": 2
    },
    "env": {}
}
```
</details>

```js
var foo = ['a', 'b', 'c'];

var x = foo[1]
var [y] = foo

console.log(x) // 'b'
console.log(y) // 'a'
```

### What did you expect to happen?

`prefer-destructuring` should not trigger when accessing array indices of > 0

### What actually happened?

`prefer-destructuring` is a false positive when accessing array indices of > 0.
When destructuring happens, (ie. `--fix`),  the first element is taken rather than the desired index.

### Participation

- [ ] I am willing to submit a pull request for this issue.

I have yet to explore how to modify rules

**Comment by cgarrovillo** [comment]
Didn't know that array destructuring isn't autofixable!

The problem stemmed from doing a `'some string'.split(' ')[1]`,  but ESL was not aware of what `.split` is (any type)
